packages/eslint-plugin/src/rules/no-redundant-type-constituents.ts

```diff
@@ -323,8 +327,58 @@ export default createRule({
               }
             }
           }
-        }
 
+          typePartFlags.length >= 2
+            ? seenUnionTypes.set(typeNode, typePartFlags)
+            : null;
+        }
+        const checkIfUnionsAreAssignable = (): undefined => {
+          let typeFlagsOfUnions: ts.TypeFlags[] = [];
+          let result = true;
+          let primitiveUnit: number;
```

Comment:
[Refactor] Right now this function stores two pieces of state: `result` (`boolean`) and `primitiveUnit` (`number`). If one is truthy, the other must be. That's a bit of a code smell for duplicate state.

Tests all still pass locally if I remove `result` altogether, set `primitiveUnit`'s type to `number | undefined`, and change from `if (result)` to `if (primitiveUnit)`.

Also, there's no need to declare `primitiveUnit` outside the `forEach`. Each iteration of the function has its own idea of what the primitive unit should be.

Proposal: delete `result` and move `primitiveUnit` inside the `forEach`?